src/backend/src/services/projects.services.ts

```diff
@@ -829,4 +829,84 @@ export default class ProjectsService {
 
     return newMaterialType;
   }
+
+  /**
+   * Update a material
+   * @param submitter the submitter of the request
+   * @param materialId the material id of the new material
+   * @param name the name of the new material
+   * @param status the status of the new material
+   * @param manufacturerName the manufacturerName of the new material
+   * @param manufacturerPartNumber the manufacturerPartNumber of the new material
+   * @param quantity the quantity of the new material
+   * @param price the price of the new material
+   * @param subtotal the subtotal of the new material
+   * @param linkUrl the linkUrl of the new material
+   * @param notes the notes of the new material
+   * @param unitName the unit name of the new material
+   * @param assemblyId the assembly id of the new material
+   * @param pdmFileName the pdm file name of the new material
+   * @throws if permission denied or material's wbsElement is undefined/deleted
+   * @returns the updated material
+   */
+  static async editMaterial(
+    submitter: User,
+    materialId: string,
+    name: string,
+    status: Material_Status,
+    manufacturerName: string,
+    manufacturerPartNumber: string,
+    quantity: number,
+    price: number,
+    subtotal: number,
+    linkUrl: string,
+    notes: string,
+    unitName?: string,
+    assemblyId?: string,
+    pdmFileName?: string
+  ): Promise<Material> {
+    const material = await prisma.material.findUnique({
+      where: {
+        materialId
+      }
+    });
+
+    if (!material) throw new NotFoundException('Material', materialId);
+    if (material.dateDeleted) throw new DeletedException('Material', materialId);
+
```

Comment:
you also need all these checks from the createMaterialEndpoint, I'd recommend putting them into a helper method or smthn:
```
    if (assemblyId) {
      const assembly = await prisma.assembly.findFirst({ where: { assemblyId } });
      if (!assembly) throw new NotFoundException('Assembly', assemblyId);
    }

    const materialType = await prisma.material_Type.findFirst({
      where: { name: materialTypeName }
    });
    if (!materialType) throw new NotFoundException('Material Type', materialTypeName);

    const manufacturer = await prisma.manufacturer.findFirst({
      where: { name: manufacturerName }
    });
    if (!manufacturer) throw new NotFoundException('Manufacturer', manufacturerName);

    const unit = await prisma.unit.findFirst({
      where: { name: unitName }
    });
    if (!unit) throw new NotFoundException('Unit', unitName);
```